### ISSUE TITLE
Support for managed expo projects

### ISSUE THREAD

**Issue by alephmelo** [Shopify/flash-list#480]
Hi there 👋 

I'm fairly new to react-native development, so I've been using the [managed](https://docs.expo.dev/workflow/using-libraries/#determining-third-party-library-compatibility) expo workflows for my toy projects. 

This is the issue I'm bumping into: 

`Invariant Violation: requireNativeComponent: "AutoLayoutView" was not found in the UIManager.`

```
$ expo --version
5.4.12

$ node --version
v16.14.2
```

```
// package.json
...
"dependencies": {
    "@react-navigation/native": "^6.0.10",
    "@react-navigation/native-stack": "^6.6.2",
    "@shopify/flash-list": "^1.0.2",
    "expo": "~45.0.0",
```

I'm assuming I'd be able to eject expo and then run `pod install` in the `ios` folder, but it would be interesting to know if I'm missing some steps on how to link to my managed expo project without having to eject.

Thanks! 🙏

**Comment by rshaul** [comment]
Haven't tried it yet but found this tweet from someone on the expo team

```
expo install @shopify/flash-list expo-dev-client
𝚎𝚡𝚙𝚘 𝚛𝚞𝚗:𝚒𝚘𝚜 - or - 𝚎𝚊𝚜 𝚋𝚞𝚒𝚕𝚍
```

https://twitter.com/Baconbrix/status/1542568635209744384

**Comment by vbylen** [comment]
You'll need to first build the expo-dev-client with @shopify/flash-list installed.

**Comment by nandorojo** [comment]
Yeah exactly. You can't use this in Expo Go, just like any libraries with native code. You'll need to create a custom dev client and use EAS to build.

**Comment by EvanBacon** [comment]
Here's an example of creating a custom dev client that uses [flash list](https://github.com/EvanBacon/Expo-Flash-List-Demo) for iOS and web (android pending an upstream Kotlin issue).

**Comment by brentvatne** [comment]
this should just work out of the box on sdk 45 projects with expo-dev-client on ios.

for android, once https://github.com/Shopify/flash-list/pull/481 lands it'll also "just work"

**Comment by fortmarek** [comment]
Thanks for the report and the [fix](https://github.com/Shopify/flash-list/pull/481) from @brentvatne 💯 

We have released a new patch version witch [1.0.3](https://github.com/Shopify/flash-list/releases/tag/v1.0.3) that should enable everyone to run Android in the expo-managed setup 🎉 

I have also updated the installation instructions to include `expo-dev-client`. I'm going to close this issue, we can reopen it if you still experience issues.
